packages/application-generic/src/services/cache/cache.service.ts

```diff
@@ -219,6 +228,25 @@ export class CacheService implements ICacheService {
 
     return addJitter(seconds, this.TTL_VARIANT_PERCENTAGE);
   }
+
+  public async sadd<TData extends string | number | Buffer>(
+    key: string,
+    ...members: TData[]
+  ): Promise<number> {
+    return this.client?.sadd(key, ...members);
+  }
+
+  public async eval<
+    TArgs extends (string | Buffer | number)[],
+    TData = unknown
+  >(script: string, keys: string[], ...args: TArgs): Promise<TData> {
+    return this.client?.eval(
+      script,
+      keys.length,
+      ...keys,
+      ...args
+    ) as Promise<TData>;
+  }
```

Comment:
Typesafe adaptator for a "generic" Redis `eval` method to the `ioredis` method. Notably, the `ioredis` library requires the length of keys (`keys.length`) to be passed as a parameter whereas many other libraries do not.